planner/core/logical_plan_builder.go

```diff
@@ -193,7 +193,9 @@ func (b *PlanBuilder) buildResultSetNode(ctx context.Context, node ast.ResultSet
 			if name.Hidden {
 				continue
 			}
-			name.OrigTblName = name.TblName
+			if name.OrigTblName.L == "" {
```

Comment:
If we set it here.
Still use `create table t4(a int default 4)`.
We would get `select default(a) from (select a from (select 1 as a) t4) t4;`'s result is `4` instead of `0`.

Reply:
```
mysql> select default(a) from (select a from (select 1 as a) t3) t4;
Field   1:  `default(a)`
Catalog:    `def`
Database:   ``
Table:      `t3`
Org_table:  ``
Type:       LONG
Collation:  binary (63)
Length:     1
Max_length: 1
Decimals:   0
Flags:      NOT_NULL NUM
```
In MySQL, it also doesn't set it if it's a virtual table.
It makes sense.

Reply:
@winoros Comment addressed, PTAL